v21.2/build-a-nodejs-app-with-cockroachdb-knexjs.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Build a Simple CRUD Node.js App with CockroachDB and Knex.js
+summary: Learn how to use CockroachDB from a simple CRUD application that uses the Knex.js query builder.
+toc: true
+twitter: false
+---
+
+<div class="filters filters-big clearfix">
+    <a href="build-a-nodejs-app-with-cockroachdb.html"><button class="filter-button">Use <strong>node-postgres</strong></button></a>
+    <a href="build-a-nodejs-app-with-cockroachdb-sequelize.html"><button class="filter-button">Use <strong>Sequelize</strong></button></a>
+    <a href="build-a-nodejs-app-with-cockroachdb-knexjs.html"><button class="filter-button current">Use <strong>KnexJS</strong></button></a>
+    <a href="build-a-typescript-app-with-cockroachdb.html"><button class="filter-button">Use <strong>TypeORM</strong></button></a>
+</div>
+
+This tutorial shows you how build a simple Node.js application with CockroachDB and [Knex.js](https://knexjs.org/).
+
+## Step 1. Start CockroachDB
+
+<div class="filters clearfix">
+  <button class="filter-button page-level" data-scope="cockroachcloud">Use {{ site.data.products.serverless }}</button>
+  <button class="filter-button page-level" data-scope="local">Use a Local Cluster</button>
+</div>
+
+<section class="filter-content" markdown="1" data-scope="cockroachcloud">
+
+### Create a free cluster
+
+{% include cockroachcloud/quickstart/create-a-free-cluster.md %}
+
+### Set up your cluster connection
+
+1. Navigate to the cluster's **SQL Users** page, and create a new user, with a new password.
+
+1. Navigate to the **Cluster Overview page**, select **Connect**.
+
+1. Take note of the connection information. You'll use it to connect to the database later in this tutorial.
```

Comment:
I actually think we should move this into create-a-free-cluster.md and emphasize the bit about the password not being available, but that's outside the scope of this ticket. Is it fair to assume that users should already know to copy their connection information down prior to this point?

Reply:
I think the connection information gathering is separate from `create-a-free-cluster.md` for a reason. Namely, not all connection flows are the same.

For example, `_includes/cockroachcloud/quickstart/set-up-your-cluster-connection.md` instructs the user to download a CA cert, but a local CA cert is actually not necessary for this particular driver. I created a new include file for that alternate workflow (`cockroachcloud/quickstart/set-up-your-cluster-connection-no-cert.md`) and used it here. That include also mentions the password bit.